File: aat-lib/src/main/java/ch/bailu/aat_lib/xml/parser/util/AbsScanner.java
package ch.bailu.aat_lib.xml.parser.util;


import java.io.IOException;

public abstract class AbsScanner {

    private final SimpleStringReader stringReader = new SimpleStringReader();
    private final Stream stringStream = new Stream(stringReader);

    public abstract void scan(Stream stream) throws IOException;


    public void scan(String string) throws IOException {
        stringReader.setString(string);
        scan(stringStream);
    }
}

File: aat-lib/src/main/java/ch/bailu/aat_lib/xml/parser/util/AbsScanner.kt
package ch.bailu.aat_lib.xml.parser.util

import java.io.IOException

abstract class AbsScanner {
    private val stringReader = SimpleStringReader()
    private val stringStream = Stream(stringReader)
    @Throws(IOException::class)

    abstract fun scan(stream: Stream)

    @Throws(IOException::class)
    fun scan(string: String) {
        stringReader.setString(string)
        scan(stringStream)
    }
}
